### ISSUE TITLE
[a11y]: Slider - Playground -Screen Reader is not announcing the Disable Status for the Slider Control

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16633]
### Package

@carbon/react

### Browser

Safari

### Package version

v1.58.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

When the Disable prop is set to true in the Playground, the screen reader does not announce the unavailable status for the slider control. However, it correctly announces the unavailable status for the Lower Bound Edit Button

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-slider--playground&args=disabled:!true

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
